packages/docregistry/src/default.ts

```diff
@@ -487,6 +488,25 @@ export abstract class ABCWidgetFactory<
     return widget;
   }
 
+  /**
+   * Identifier of the content provider required for the widget (if any).
+   *
+   * Throws Error if the content provider was already set.
+   *
+   * @experimental
+   */
+  get contentProviderId(): string | undefined {
```

Comment:
For reference , this seems to be causing the following issue downstream in Notebook: https://github.com/jupyter/notebook/pull/7559#issuecomment-2573550718

```
Error: ../../node_modules/@jupyterlab/docregistry/lib/default.d.ts(267,9): error TS2380: The return type of a 'get' accessor must be assignable to its 'set' accessor type
  Type 'undefined' is not assignable to type 'string'.
```